docs/Transport Manager Programming/index.md

```diff
@@ -1,567 +1,407 @@
 
-```
+**OnSendFail**
 
-Create a class for the device that will be used by Controller to manage devices and implement all virtual methods
+Indicates that a Transport Adapter failed to send a message properly. This could prompt the transport manager to take action that would reconcile the errors. This event is not currently raised to the Transport Manager Listeners.
 
-```
-#include "transport_adapter_impl.h"
+**OnReceivedDone**
 
-class Mydevice : public Device
-{
-    virtual ~Device();
-    virtual bool isSameAs(const Device *other_device) const;
-    virtual ApplicationList getApplicationList() const;
-}
+Indicates that a Transport Adapter has successfully received a message. The received data is eventually passed to Transport Manager Listeners which will process that data.
 
-```
+**OnReceivedFail**
 
-Create the necessary worker classes by deriving the appropriate basic worker and fill in the necessary functionality
+Indicates that a Transport Adapter has failed to properly receive a message.
 
-```
-#include "transport_adapter_impl.h"
-class MyDeviceScanner : public DeviceScanner
-{
+**OnUnexpectedDisconnect**
 
-}
-class MyServerConnectionFactory : public ServerConnectionFactory
-{
+Indicates that a device has been unexpectedly disconnected. This event could be emitted in the case of a device being disconnected or a connection being aborted. This event will prompt the Transport Manager to remove the disconnected connection.
 
-}
-class MyClientListener : public ClientConnectionListener
-{
+**OnTransportSwitchRequested**
 
-}
+Indicates that a transport switch has been requested. This will prompt the Transport Manager to begin transport switching.
 
-```
+**OnTransportConfigUpdated**
 
-These workers should use Connection and Device which were created in previous steps. For instance, scanner should add a list of devices to controller. This list will be used later when the `connect` request is received. To create a data path the connection class should be used. When the connection starts it will update Controller with the pointer to this connection.
+Indicates that the Transport Config has been updated. This will prompt the Protocol Handler (a Transport Manager Listener) to check for updates to things like the TCP listening address and port.
 
-When everything is created it is time to combine everything.
+**OnConnectPending**
 
-```
-#include "transport_adapter_impl.h"
-
-MyTransportAdapter::MyTransportAdapter() : Transport AdapterImpl(
-    new MyDeviceScanner(),
-    new MyServerConnectionFactory(),
-    new MyClientListener()
-)
-{
-}
-```
+Indicates that a connection is pending. The Transport Manager will then add the connection to the connection list if it has not already been added. In the case of the cloud websocket transport adapter, this event is emitted once a connection with a new device is created.
```

Comment:
> In the case of the cloud websocket transport adapter, this event is emitted once a connection with a new device is created.

Technically nothing "connected" yet. Connection wont happen until the user activates the app, which requires a different TA event. 

Change to something like:
> In the case of the cloud websocket transport adapter, this event is emitted once a connection configuration is known to SDL Core before the connection is actually established.